### DISCUSSION TITLE
Target New old values when working with Audit.Webapi

### DISCUSSION THREAD

**Discussion by TahaLeo** [thepirat000/Audit.NET#533] [Show and tell]
Is there any way to save old and new values using Audit.WebApi,
can anyone share the flow for that

**Comment by thepirat000** [comment]
Please provide more information or an example of what are you looking for. What kind of values do you mean to save old and new?

**Reply by TahaLeo** [reply]
we are calling API to update data, now what we are looking for is to save old values prior to update and new values at the time of update, Is there any provision at API controller level.

**Comment by TahaLeo** [comment]
Hi @thepirat000,
Can you please help me out with how i can log multiple entities with same method, currently am using following

       var auditScope = this.GetCurrentAuditScope();
             auditScope.Comment(Input.comment);
       
        var auditScope1 = this.GetCurrentAuditScope();
            auditScope1.Comment(Input.comment);
            
 But with above code the issue is am able to log only one entity if i try to call it again, it will only save the auditscope1 and will ignore auditScope    
 
 how can i save the auditScope and auditScope1 both within same method

**Comment by thepirat000** [comment]
Audit.WebApi creates one audit scope/audit event per HTTP call. It doesn't maintain any status between different calls, so the only way to co-relate these audit events is at the data level. 

If you need old/new values in the same audit event when updating an entity, then you should probably do the audit at the data access layer when updating the entity (i.e. using EntityFramework.Core + [Audit.EntityFramework.Core](https://github.com/thepirat000/Audit.NET/blob/master/src/Audit.EntityFramework/README.md))
